_posts/2020-07-12-amiko-hydras-proof.markdown

```diff
@@ -0,0 +1,598 @@
+We define a function \\\(K\\\) which is like \\\(M\\\) except its output is a tuple, with \\\(M\\\) then defined as \\\(M(A) = P(K(A))\\\).
+Let \\\(\tau(\alpha, m) = (\forall A \in \mathbb{H}, Q(A) \le m \implies ( M(A) = \alpha \implies \pi(A)) )\\\)
+We also fix the definition of hydra comparison as \\\((A \lesseqgtr B) = (M(A) \lesseqgtr M(B))\\\),
+which is needed because the original post on the hydras did not explicitly give a comparison algorithm.
+
+*In words, \\\(\tau\\\) is the predicate "for all hydras which correspond to this ordinal, that hydra terminates".
+There may be no such hydras, in which case it is vacuously true.*
+
+**3. Definition of sub-expression comparison.**
+Detine \\\(a \lessapprox b \iff a \lessapprox^\emptyset b\\\)
+Define \\\(a \lessapprox_=^c b \iff a = b \lor a \lessapprox^c b\\\)
+Define \\\(a \lessapprox^c b \iff b \neq 0 \land (a = 0 \lor a \neq 0 \land (a \lessapprox^{c\cup \\\{b\\\}} b[2]\\\)
+\\\(\lor (a[2] \lessapprox_=^{c\cup \\\{b\\\}} b[2] \lor a[2] \lessapprox_=^c b) \land (\\\)
+\\\(a[0] \lessapprox^{c\cup \\\{b\\\}} b[0] \land (a[1] \lessapprox_=^{c\cup \\\{b\\\}} b[1] \lor \exists x \in c\cup \\\{b\\\}, a[1]\lessapprox x)\\\)
+\\\(\lor a[0] \lessapprox_=^{c\cup \\\{b\\\}} b[0] \land a[1] \lessapprox^{c\cup \\\{b\\\}} b[1] )\\\)
+\\\(\lor a[0] \lessapprox_=^{c\cup \\\{b\\\}} b[0] \land a[1] \lessapprox_=^{c\cup \\\{b\\\}} b[1] \land a[2] \lessapprox^{c\cup \\\{b\\\}} b[2] ))\\\)
+Note that \\\(c_0 \subset c_1 \implies (a \lessapprox^{c_0} b \implies a \lessapprox^{c_1} b)\\\), so it is safe to drop \\\(c\\\) in certain contexts, as is done in the proof.
+
+*The clause \\\(\exists x \in c\cup \\\{b\\\}, a[1]\lessapprox x\\\) permits a [1]-term to increase when the [0]-term decreases,
+but to prevent non-terminating stacking, we still need the [1]-term to be less than something else - to be precise, it must be less than some parent expression.*
+
+**4. Hydra step always decreases the ordinal or a sub-expression, given that label hydras also decrease.**
+\\\(\forall A \in (\mathbb{H} - \\\{0_H\\\}), (\forall B \in A, K(S(B)) \lessapprox K(B)) \implies K(S(A)) \lessapprox K(A)\\\)
+Note that \\\(S\\\) is nondeterministic due to the free variable \\\(N\\\), so there is an implied \\\(\forall N \in \mathbb{N}\\\).
+
+**5. Definition of hydra depth.**
+\\\(Q(A) = \max \\\{ 0 \\\} \cup \\\{ Q(B) + 1 | B \in A \\\} \\\)
+
+*The depth of \\\(0_H\\\) is \\\(0\\\), and the depth of all other hydras is \\\(1\\\) more than the highest depth of any label.*
+
+**6. Transfinite induction for termination of hydras with limited depth.**
+We define \\\(R(m) = (\forall \alpha \in \chi, \tau(\alpha, m))\\\)
+Then,
+\\\(\tau(0, m) \land (\forall \alpha \in \chi, (\forall \beta \in \alpha, \tau(\beta, m)) \implies \tau(\alpha, m)) \implies R(m)\\\)
+
+*In words, if it holds for \\\(0\\\), and if it holds for some \\\(\alpha\\\) if it holds for all \\\(\beta\\\) below \\\(\alpha\\\), then it holds for all ordinals.
```

Comment:
Do you mean to say "In other words"?

Reply:
No, I mean "in words". Symbols are hard to understand and not the universal language of math they're sometimes made out to be. I think expressing the same idea again in words helps to avoid confusion.